### ISSUE TITLE
Wrong hint in airplane mode

### ISSUE THREAD

**Issue by mhettig** [martinhellwig/DatapassWidget#15]
When entering airplane mode the "carrier not supported" hint appears which is technically wrong, respectively we cannot make a statement about the current carrier in this situation. Therefore we should either:
- Show the "turn on mobile data" hint
- Or behave like WiFi is turned on (e.g. keep all information and grayish the progressbar) <- would be more consistent

**Comment by martinhellwig** [comment]
Closed with commit 8382187

carrier for each widget is stored now, so we also know in airplane-mode, which carrier the user had chosen

**Comment by mhettig** [comment]
Good work. But a small bug still appears: If the widget is dragged onto the homescreen for the first time and the phone is in airplane mode while doing so, the "carrier is not supported" hint is shown. Although it's correct that we cannot know about the carrier at this time the "please turn on mobile data" hint would be more target-aimend.

**Comment by martinhellwig** [comment]
To the first point: Newest commit now sets CARRIER_NOT_SELECTED flag, when adding a widget while in airplane-mode. So onClick the app will try again to get the wanted carrier (or starts a dialogue, if multiSim-phone). This will be done, till a carrier is chosen for the widget.

Second point: This only applies for old widgets, which were created before the commit 8382187, because at this stage the carrier was not stored for each widget. Simply clear appData and alls hould work fine.

Last point: This should not be the case. On my phone, the widget gets updated after switching from airplane-mode to normal/Wifi

**Comment by mhettig** [comment]
Hm, unfortunately I cannot agree with you. What I've done is:

1. Uninstalled the current version (and remove Widget from homescreen, which seems a special behavior of the Huawei Launcher - see #1)
2. Switch to airplane mode (before that WiFi and Mobile Data were turned on)
3. Install latest version 95364f8 via Android Studio
4. Drag widget onto the homescreen
5. **Result 1:** "Carrier not suported" hint is shown. So the behavior mentioned above (first point) still applies
6. Switch off airplane mode (after at least waiting the threshold of 5 seconds)
7. Phone seems to connect to Mobile Data for 1-2 seconds before WiFi connection is established
8. **Result 2:** No auto update is triggered (third point) and therefore "carrier not supported" hint is still shown
9. Press the widget for manual update
10. **Result 3:** Loading animation is shown resulting in the "carrier not supported" hint
11. Switching off WiFi
12. Auto update is triggered (with loading animation)
13. **Result 4:** "Carrier not supported" hint is still shown (second point)
14. Press the widget again for manual update, outcome like Result 4

**Comment by mhettig** [comment]
I've checked the newest commit. Here are my results:

1. Uninstalled current version
2. Switch to airplane mode (before that WiFi and Mobile Data were turned on)
3. Install latest version 8da0ed234d1a086deb308ba6d1001c791fc3adda via Android Studio
4. Drag widget onto the homescreen
5. **Result 5:** The normal loading animation is shown resulting in the "Turn on Mobile Data" hint -> good
6. Strangely shortly thereafter another auto update is triggered
7. **Result 6:** After loading animation is finished the "Turn on Mobile Data" hint is shown again
8. Switch off airplane mode (after at least waiting the threshold of 5 seconds)
9. Phone seems to connect to Mobile Data for 1-2 seconds before WiFi connection is established
10. Auto update is triggered but it seems to be the "ultra silent" one
11. **Result 7:** "Carrier not selected" hint is shown - which should never happen on single sim phones
12. Press the widget for manual update
13. **Result 8:** After loading animation is finished "Carrier not selected" hint is shown.
14. After that (like in step 6) another auto update is triggered (again with no user interaction)
15. **Result 9:** After loading animation is finished "Switch off WiFi" hint is shown (which is technically correct)
16. Switching off WiFi
17. Auto update is triggered (with loading animation)
18. **Result 10:** Correct values from the homepage are shown -> good

**Comment by martinhellwig** [comment]
Yep, same on my phone while testing. Point 11 is wanted from me, because the internal state of the widget is CARRIER_NOT_SELECTED, so it shortly appear as that.

But all in all the multiple reloads should not appear. Now its time to get rid off them. If we do it right, the CARRIER_NOT_SELECTED hint will never be shown
